sql/001-create-table-and-load-data.sql

```diff
@@ -1,16 +1,15 @@
-DROP TABLE IF EXISTS money_diarys;
+DROP TABLE IF EXISTS money_diaries;
 
-CREATE TABLE money_diarys (
+CREATE TABLE money_diaries (
  id int unsigned AUTO_INCREMENT,
- usingday DATE,
- category VARCHAR(100) NOT NULL,
- amount int,
- comment VARCHAR(100)NOT NULL,
-
+ DATE_ADDED DATE,
+ INCOME VARCHAR(100) NOT NULL,
+ EXPENSE VARCHAR(100) NOT NULL,
+ AMOUNT int,
+ COMMENT VARCHAR(100)NOT NULL,
  PRIMARY KEY(id)
 );
 
-INSERT INTO money_diarys (usingday, category, amount, comment) VALUES ('2024-04-12', 'food', 3500, 'good taste');
-INSERT INTO money_diarys (usingday, category, amount, comment) VALUES ('2024-04-12', 'transportation', 188, 'shinjuke-nearstation');
-
-
+INSERT INTO money_diaries (DATE_ADDED, INCOME, EXPENSE, AMOUNT, COMMENT) VALUES ('2024-04-12', 'なし', '夜ごはん', -6000, '新宿');
```

Comment:
テーブル設計悩まれましたね。
いいこと。
`'なし'`についてですがこれは実質データがない NULL の状態ですよね。
```suggestion
INSERT INTO money_diaries (DATE_ADDED, INCOME, EXPENSE, AMOUNT, COMMENT) VALUES ('2024-04-12', NULL, '夜ごはん', -6000, '新宿');
```
NULLを許容してもいいのですがNULLはデータがないことをしめす = 不正なデータなのであまり許容したくはないです。

たとえば、収入か支出かを表すカラムを使うのはどうでしょうか？

| 列名     | データ型   | 説明                           |
|----------|------------|--------------------------------|
| id       | INTEGER    | レコードの一意の識別子         |
| 記録日   | DATE       | 収支が発生した日付             |
| 収支種別 | VARCHAR    | 収入か支出かを示す             |
| 項目名   | VARCHAR    | 支出や収入の種類を示す         |
| 金額     | DECIMAL    | 金額（通貨の小数点以下の桁数を考慮）|
| 備考     | TEXT       | その他の情報やコメント           |

Reply:
遅くなってしまいすみません。
先生のアドバイスから、nullはよくないと感じたので、収入、出費を先に出力、その後、項目名を入力できる形に変更することにしました。
確認お願いします！